firefly/main.py

```diff
@@ -1,38 +1,60 @@
+import os
 import sys
 import argparse
 import importlib
+import yaml
 from .app import Firefly
 from .server import FireflyServer
+from .validator import ValidationError, FireflyError
 
 def parse_args():
     p = argparse.ArgumentParser()
     p.add_argument("-b", "--bind", dest="ADDRESS", default="127.0.0.1:8000")
-    p.add_argument("functions", nargs='+', help="functions to serve")
+    p.add_argument("-c", "--config", dest="config_file", default=None)
+    p.add_argument("functions", nargs='*', help="functions to serve")
     return p.parse_args()
 
-def load_function(function_spec):
+def load_function(function_spec, path=None, name=None):
     if "." not in function_spec:
         raise Exception("Invalid function, please specify it as module.function")
 
     mod_name, func_name = function_spec.rsplit(".", 1)
     mod = importlib.import_module(mod_name)
     func = getattr(mod, func_name)
-    return (func_name, func)
+    return ((path or "/"+func_name), (name or func_name), func)
```

Comment:
The line above is confusing. It would be better to split that into multiple lines to make the intent clear. 

    path = path or "/" + func_name
    name = name or func_name
    return (path, name, func)